### PR TITLE
Feature/generic saga error handler

### PR DESCRIPTION
**Pull request by ProH4Ck**
Implements #807

### CODE REVIEW

File: Source/EventFlow/Sagas/SagaErrorHandler.cs [hunk 1]
```diff
@@ -29,14 +29,29 @@ namespace EventFlow.Sagas
...
+
         public Task<bool> HandleAsync(
             ISagaId sagaId,
             SagaDetails sagaDetails,
             Exception exception,
             CancellationToken cancellationToken)
         {
-            // The default handler cannot handle anything!
+            // Search for a specific SagaErrorHandler<Saga> based on saga type
+            ISagaErrorHandler sagaErrorHandler = _sagaErrorHandlerFactory(sagaDetails.SagaType);
```
Comment (rasmus) [inline Source/EventFlow/Sagas/SagaErrorHandler.cs]: @ProH4Ck I think this should be moved to the `DispatchToSagas` instead. Doing it here makes it harder to do a "catch all" error handler as well as you will need to copy the code to handle the specific error handlers yourselv.

**Comment by rasmus** [comment]
@ProH4Ck ❤️ thanks for the PR

File: Source/EventFlow/Sagas/SagaErrorHandler.cs (hunk 1 above)
Comment (ProH4Ck) [reply Source/EventFlow/Sagas/SagaErrorHandler.cs]: Correct. I'll push changes to the PR in the evening

**Comment by rasmus** [comment]
❤️

**Pull request merged**